src/v/archival/tests/service_fixture.h

```diff
@@ -171,6 +176,12 @@ class archiver_fixture
     void add_topic_with_random_data(const model::ntp& ntp, int num_batches);
     /// Provides access point for segment_matcher CRTP template
     storage::api& get_local_storage_api();
+    /// Get archival scheduler service
+    archival::internal::scheduler_service_impl& get_scheduler_service() {
+        auto p = reinterpret_cast<archival::internal::scheduler_service_impl*>(
+          &app.archival_scheduler.local());
```

Comment:
Why `reinterpret_cast`? I think this is simply a cast to a base class so could be a static or even implicit cast?

Reply:
yeh let's try to avoid reinterpret_cast

Reply:
Yep, makes sense

Reply:
actually, the base class is private so `static_cast` is not possible here